en/sql/function/json_fn.rst

```diff
@@ -631,3 +631,119 @@ Note that json objects do not check containment the same way json arrays do. It
       json_contains('["a", "b1", ["a", {"k":["b2"]}]]','["b1", {"k":"b2"}]')
     ========================================================================
                                                                            1
+
+JSON_MERGE_PATCH
+===================================
+
+.. function:: JSON_MERGE_PATCH (json_doc, json_doc [, json_doc] ...)
+
+The **JSON_MERGE_PATCH** function merges 2 or more json docs and returns the resulting merged json. **JSON_MERGE_PATCH** differs from **JSON_MERGE_PRESERVE** in that it will take the second argument when encountering merging conflicts. **JSON_MERGE_PATCH** is compliant with
+`RFC 7396 <https://tools.ietf.org/html/rfc7396/>`_.
+
+The merging of 2 json documents is performed with the following rules, recursively:
+
+- when first argument is not an object, the result of the merge is the second object. As an exception, when the second argument is an object, the result of the merge is the merge result of the second argument with an empty object.
```

Comment:
Might be easier to understand if this line is split into subcases - no argument is json object, first is a json object, second is not, first is not, second is.